R/estimate_KM.R

```diff
@@ -86,39 +82,51 @@ estimate_KM <- function(
   if (!is.data.frame(data))
     stop("Data does not have class `data.frame`.")
 
-  reqcols <- c(strata, CNSR, AVAL)
-  if (! all(reqcols %in% colnames(data))){
-    stop(paste0("Following columns are missing from `data`: ", paste(setdiff(reqcols, colnames(data)), collapse = " "), "."))
+  if (is.null(formula)) {
+    reqcols <- c(strata, CNSR, AVAL)
+    if (! all(reqcols %in% colnames(data))){
+      stop(paste0("Following columns are missing from `data`: ", paste(setdiff(reqcols, colnames(data)), collapse = " "), "."))
+    }
+
+    if (!is.numeric(data[[AVAL]])){
+      stop("Analysis variable (AVAL) is not numeric.")
+    }
+
+    if (!is.numeric(data[[CNSR]])){
+      stop("Censor variable (CNSR) is not numeric.")
+    }
+  }
+  else if (!inherits(formula, "formula")) {
+    stop("Argument `formula=` must be class 'formula'.")
```

Comment:
Can you include a requirement test for this: 1. That we allow for the argument. That the argument is tested, the argument cancels out aval and cnsr, ...

Reply:
added

Reply:
I'll recheck later this week.